File: 08 Data Import and Export/ExportFlatPattern/SolidWorksMacro.vb
Imports SolidWorks.Interop.sldworks
Imports SolidWorks.Interop.swconst
Imports System.Runtime.InteropServices
Imports System


Partial Class SolidWorksMacro
    Public Sub main()
        'get the active document
        Dim swDoc As ModelDoc2 = swApp.ActiveDoc
        If swDoc.GetType <> swDocumentTypes_e.swDocPART Then
            MsgBox("This macro is for parts only.", MsgBoxStyle.Exclamation)
            Exit Sub
        End If
        'get the part interface
        Dim swPart As PartDoc = swDoc
        'set a new path for the flat pattern
        Dim flatPatternPath As String
        Dim ext As String = ".DXF"
        'could also use .DWG if prefered
        flatPatternPath = IO.Path.ChangeExtension(swDoc.GetPathName, ext)
        'export the flat pattern with bend lines
        Dim bendSetting As Long
        bendSetting = 2 ^ 0 + 2 ^ 2
        'bendSetting = 2 ^ 0  'use this setting for no bend lines
        swPart.ExportToDWG2(flatPatternPath, swDoc.GetTitle,
                            swExportToDWG_e.swExportToDWG_ExportSheetMetal,
                            True, Nothing, False, False, bendSetting, Nothing)
    End Sub
    ''' <summary>
    ''' The SldWorks swApp variable is pre-assigned for you.
    ''' </summary>
    Public swApp As SldWorks
End Class

File: C# Examples/08 Data Import and Export/ExportFlatPattern/SolidWorksMacro.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Forms;

using SolidWorks.Interop.sldworks;
using SolidWorks.Interop.swconst;


namespace ExportFlatPattern
{
    public partial class SolidWorksMacro
    {
public void Main()
{
    ModelDoc2 swDoc = null;
    PartDoc swPart = null;

    //get the active document
    swDoc = (ModelDoc2)swApp.ActiveDoc;
    //exit if not a part
    if (swDoc.GetType() != (int)swDocumentTypes_e.swDocPART)
    {
        swApp.SendMsgToUser2("This macro is for parts only.", 
            (int)swMessageBoxIcon_e.swMbWarning, (int)swMessageBoxBtn_e.swMbOk);
        return;
    }
    //get the part interface
    swPart = (PartDoc)swDoc;
    //set a new path for the flat pattern
    string FlatPatternPath;
    string ext = ".DXF";
    //could also use .DWG if prefered
    FlatPatternPath = System.IO.Path.ChangeExtension(swDoc.GetPathName(), ext);
    //export the flat pattern with bend lines
    int bendSetting = 2 ^ 0 + 2 ^ 2;
    //int bendSetting = 2 ^ 0 //use this setting for no bend lines
    swPart.ExportToDWG2(FlatPatternPath, swDoc.GetTitle(), 
        (int)swExportToDWG_e.swExportToDWG_ExportSheetMetal,
        true, null, false, false, bendSetting, null);
    return;
}

        // The SldWorks swApp variable is pre-assigned for you.
        public SldWorks swApp;

    }
}

